src/test/java/com/pubnub/contract/subscribe/steps/WhenSteps.kt

```diff
@@ -0,0 +1,69 @@
+package com.pubnub.contract.subscribe.steps
+
+import com.pubnub.api.PubNub
+import com.pubnub.api.callbacks.SubscribeCallback
+import com.pubnub.api.models.consumer.PNStatus
+import com.pubnub.api.models.consumer.objects_api.channel.PNChannelMetadataResult
+import com.pubnub.api.models.consumer.objects_api.membership.PNMembershipResult
+import com.pubnub.api.models.consumer.objects_api.uuid.PNUUIDMetadataResult
+import com.pubnub.api.models.consumer.pubsub.PNMessageResult
+import com.pubnub.api.models.consumer.pubsub.PNPresenceEventResult
+import com.pubnub.api.models.consumer.pubsub.PNSignalResult
+import com.pubnub.api.models.consumer.pubsub.files.PNFileEventResult
+import com.pubnub.api.models.consumer.pubsub.message_actions.PNMessageActionResult
+import com.pubnub.contract.state.World
+import com.pubnub.contract.subscribe.state.SubscribeState
+import io.cucumber.java.en.Given
+import java.util.*
+
+class WhenSteps(
+    private val world: World,
+    private val subscribeState: SubscribeState
+) {
+
+
+    @Given("I subscribe to {string} channel")
+    fun I_subscribe_to_the_channel(channelName: String) {
+        val listener = getSubscribeCallbackImpl()
+        world.pubnub.addListener(listener)
+        world.pubnub.subscribe()
+            .channels(Collections.singletonList(channelName))
```

Comment:
```suggestion
            .channels(listOf(channelName))
```

Reply:
What is wrong with Collections.singletonList ?

Reply:
Nothing wrong per se. `listOf` is just more flexible kotlin equivalent.

Reply:
I wanted to explicitly indicate that subscribing to only one channel.